astroid/brain/brain_dataclasses.py

```diff
@@ -235,8 +228,48 @@ def _generate_dataclass_init(assigns: list[AnnAssign]) -> str:
         if not init_var:
             assignments.append(assignment_str)
 
-    params_string = ", ".join(["self"] + params)
     assignments_string = "\n    ".join(assignments) if assignments else "pass"
+    new_params_string = ", ".join(params)
+
+    try:
+        base: ClassDef = next(next(iter(node.bases)).infer())
```

Comment:
`node.bases` are `nodes.Name`. So, I get the first base node which then gets inferred, of which I get the first inference result. This should be the `ClassDef`.

This works because we mangle the `__init__` of each `dataclass` so there is no need to iterate over all base classes, the last base is fine.

I use `next(iter(` as it is a nice safe guard against not finding any bases using the `try ... except` I needed anyway for the call to `.infer()`.